Catching all exceptions raised within an enclosed computation, while
remaining responsive to (external) asynchronous exceptions.
